### ISSUE TITLE
make it possuble to just supply a full config

### ISSUE THREAD

**Issue by trondhindenes** [open-telemetry/opentelemetry-helm-charts#1160]
I'm finding it incredibly difficult to set up a custom collector config, because I need to disable the things I don't want. I can't just supply port values if I need non-standard ones either. 

Please consider making the config easier by providing an option to just supply a full config and list of ports

**Comment by JaredTan95** [comment]
have you tried `configMap.existingName`?

**Comment by TylerHelmuth** [comment]
You should be able to do

```yaml
ports:
  custom:
    enabled: true
    containerPort: 9999
    servicePort: 9999
    hostPort: 9999
    protocol: TCP
```

to add custom ports to both the container and service.  `ports` is a map so it will merge cleanly.

**Comment by TylerHelmuth** [comment]
Can you be more specific about what is difficult to configure?  You can remove default components using helm's `null` feature such as 

```yaml
config:
  receivers:
    jaeger: null
    prometheus: null
    zipkin: null
  service:
    pipelines:
      traces:
        receivers:
          - otlp
      metrics: null
      logs: null
```

Full example: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/examples/deployment-otlp-traces/values.yaml

**Comment by boxidau** [comment]
+1 on this issue. Having to null out the default config requires merging the nulled structure into our config which seems a little messy. Changes to the default config would then be breaking changes on chart updates.

Having a second alternative key `customConfig` (or similar) in values where a user can define their entire config would be much easier to deal with in cases where a user has an already well formed config they'd like to use.

The other option of an externally managed ConfigMap is okay and gives the flexibility but loses the [auto-reload functionality on config updates](https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1194)

**Comment by lerminou** [comment]
@TylerHelmuth , it seems to not work with null inside the config directly:
https://helm.sh/docs/chart_template_guide/values_files/#deleting-a-default-key

If I try on my own, following the 'otel-trace only' https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-collector/examples/deployment-otlp-traces

I tried:
```
config:
  receivers:
    jaeger: null
    prometheus: null
    zipkin: null
  service:
    pipelines:
      traces:
        receivers:
          - otlp
      metrics: null
      logs: null
```

or more specific:
```
    service:
      pipelines:
        traces:
          exporters:
            - otlp
          receivers:
            - otlp
            - zipkin
        metrics:
          exporters: {}
          processors: {}
          receivers: {}
        logs:
          exporters: {}
          processors: {}
          receivers: {}

```

in each case I have an error during the helm install:
```
coalesce.go:286: warning: cannot overwrite table with non table for otel-collector.opentelemetry-collector.config.service.pipelines.metrics (map[exporters:[debug] processors:[memory_limiter batch] receivers:[otlp prometheus]])
coalesce.go:286: warning: cannot overwrite table with non table for otel-collector.opentelemetry-collector.config.service.pipelines.metrics (map[exporters:[debug] processors:[memory_limiter batch] receivers:[otlp prometheus]])
coalesce.go:286: warning: cannot overwrite table with non table for otel-collector.opentelemetry-collector.config.service.pipelines.metrics (map[exporters:[debug] processors:[memory_limiter batch] receivers:[otlp prometheus]])

```

And the generated configmap in my cluster has all the keys.

**Comment by TylerHelmuth** [comment]
I really really don't want to have to do anything special in the chart when this is truly a helm bug.  But, considering this bug still isn't fixed, I'm willing to discuss work around solutions.

**Comment by TylerHelmuth** [comment]
Things I'd like to see in a solution:

1. non-breaking change.  
2. Maybe still works with presets

**Comment by booyaa** [comment]
Apologies for the cross-post from the fixing PR, but I figured someone else might come across this  issue and want an update.

> Good news, the upstream fix should land in Helm 3.17.1 🎉 https://github.com/helm/helm/pull/12879#issuecomment-2616724718
